lightning/src/chain/channelmonitor.rs

```diff
@@ -1763,9 +1769,23 @@ impl<ChanSigner: ChannelKeys> ChannelMonitor<ChanSigner> {
 		// Determine new outputs to watch by comparing against previously known outputs to watch,
 		// updating the latter in the process.
 		watch_outputs.retain(|&(ref txid, ref txouts)| {
-			let output_scripts = txouts.iter().map(|o| o.script_pubkey.clone()).collect();
-			self.outputs_to_watch.insert(txid.clone(), output_scripts).is_none()
+			let idx_and_scripts = txouts.iter().map(|o| (o.0, o.1.script_pubkey.clone())).collect();
+			self.outputs_to_watch.insert(txid.clone(), idx_and_scripts).is_none()
```

Comment:
Can we iterate the new watch txn to assert they're known types so that the panic!() two hunks down is definitely correct?

Reply:
Actually, its all test-only, it doesnt matter.